pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -223,6 +226,13 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		return errors.Wrapf(err, "Unable to delete network interface")
 	}
 
+	// Create public IP resource if requested
+	publicIPName := azure.GeneratePublicIPName(s.scope.Cluster.Name, fmt.Sprintf("%v-public-ip-address", s.scope.Machine.Name))
```

Comment:
Can you please create a separate function 

GenerateNodePublicIPName(NodeName)
{
    fmt.Printf("%s-publicip", nodename, hash)
}
Public ip names some restrictions, its better to keep it short as well

Reply:
+1, create `GenerateMachinePublicIPName` and replace both occurrences.